examples/bond/scripts/deploy.js

```diff
@@ -40,22 +46,26 @@ async function run(runtimeEnv, deployer) {
 		TMPL_MATURITY_DATE: Math.round(new Date().getTime() / 1000) + 240,
 	};
 	// Create Application
-	const bondAppInfo = await deployer.deployApp(
-		managerAcc,
-		{
-			appName: "BondApp",
-			metaType: types.MetaType.FILE,
-			approvalProgramFilename: "bond-dapp-stateful.py",
-			clearProgramFilename: "bond-dapp-clear.py",
-			localInts: 1,
-			localBytes: 1,
-			globalInts: 8,
-			globalBytes: 15,
-			appArgs: appArgs,
-		},
-		{},
-		placeholderParam
-	);
+	const bondAppInfo = await deployer
+		.deployApp(
+			managerAcc,
+			{
+				appName: "BondApp",
+				metaType: types.MetaType.FILE,
+				approvalProgramFilename: "bond-dapp-stateful.py",
+				clearProgramFilename: "bond-dapp-clear.py",
+				localInts: 1,
+				localBytes: 1,
+				globalInts: 8,
+				globalBytes: 15,
+				appArgs: appArgs,
+			},
+			{},
+			placeholderParam
+		)
```

Comment:
> It looks good, however I think we should rethink it. Maybe it would be better to throw the error inside of deployer? I am not sure about creating more wrappers for deployer since the examples are supposed to show real use cases that user can just adopt in their projects.

The better solution to his can be to directly throw the error in executeTx method, like I did with web pacakge, then in the examples we only need to catch it in places where the Transaction is SUPPOSED TO FAIL, and all other places if it failed it will throw an error

Reply:
can do that, but user usually deploys the app once or twice in real use case, so I thought it's better to keep it this way
What do you think?